src/colvarproxy.h

```diff
@@ -48,11 +48,53 @@ class colvarproxy_system {
   /// Destructor
   virtual ~colvarproxy_system();
 
-  /// \brief Value of the unit for atomic coordinates with respect to
-  /// angstroms (used by some variables for hard-coded default values)
-  virtual cvm::real unit_angstrom() = 0;
+  // **********************************************************************
+  // TODO FIXME unsolved: calls to back-end PBC functions assume back-end length unit!
+  // We can use different unit from back-end in VMD bc using PBC functions from colvarproxy base class
+  // or test and default to internal PBC functions if we have our own units?
```

Comment:
The needed check is between two versions of the PBC functions: one using the backend's PBC handler (or in any case, its unit cell parameters) and one using internal parameters entirely.  Since the units are (or at least should be) set at init time, I think this can be done with class polymorphism.  Objects could store a pointer to a "PBC handler" object, which has several variants, including calling the backend's function or handling the PBCs internally (with internal units).  Another variant of this object could be to just use straight differences, which would eliminate a lot of the checks for `f_cvc_pbc_minimum_image`. What do you think?

Reply:
Sorry, I introduced this note early in the process and it's outdated now. Since then the problem has been fixed in VMD by adapting our internal wrapping function, and for all other backends we use the same units internally. I'm changing this comment to reflect the current situation.